L1Trigger/L1TTrackMatch/test/Analyzer_DisplacedMuon.C

```diff
@@ -0,0 +1,3526 @@
+  tree->SetBranchAddress("trkExt_matchtp_x", &trk_matchtp_x, &b_trk_matchtp_x);
+  tree->SetBranchAddress("trkExt_matchtp_y", &trk_matchtp_y, &b_trk_matchtp_y);
+  tree->SetBranchAddress("trkExt_matchtp_z", &trk_matchtp_z, &b_trk_matchtp_z);
+  tree->SetBranchAddress("tp_pt", &tp_pt, &b_tp_pt);
+  tree->SetBranchAddress("tp_eta", &tp_eta, &b_tp_eta);
+  tree->SetBranchAddress("tp_phi", &tp_phi, &b_tp_phi);
+  tree->SetBranchAddress("tp_dxy", &tp_dxy, &b_tp_dxy);
+  tree->SetBranchAddress("tp_d0", &tp_d0, &b_tp_d0);
+  tree->SetBranchAddress("tp_z0", &tp_z0, &b_tp_z0);
+  tree->SetBranchAddress("tp_x", &tp_x, &b_tp_x);
+  tree->SetBranchAddress("tp_y", &tp_y, &b_tp_y);
+  tree->SetBranchAddress("tp_z", &tp_z, &b_tp_z);
+  tree->SetBranchAddress("tp_d0_prod", &tp_d0_prod, &b_tp_d0_prod);
+  tree->SetBranchAddress("tp_z0_prod", &tp_z0_prod, &b_tp_z0_prod);
+  tree->SetBranchAddress("tp_pdgid", &tp_pdgid, &b_tp_pdgid);
+  tree->SetBranchAddress("tp_isHToMu", &tp_isHToMu, &b_tp_isHToMu);
+  tree->SetBranchAddress("tp_isHToB", &tp_isHToB, &b_tp_isHToB);
+  tree->SetBranchAddress("tp_nmatch", &tp_nmatch, &b_tp_nmatch);
+  tree->SetBranchAddress("tp_nstub", &tp_nstub, &b_tp_nstub);
+  tree->SetBranchAddress("tp_eventid", &tp_eventid, &b_tp_eventid);
+  tree->SetBranchAddress("tp_charge", &tp_charge, &b_tp_charge);
+  tree->SetBranchAddress("matchtrkExt_pt", &matchtrk_pt, &b_matchtrk_pt);
+  tree->SetBranchAddress("matchtrkExt_eta", &matchtrk_eta, &b_matchtrk_eta);
+  tree->SetBranchAddress("matchtrkExt_phi", &matchtrk_phi, &b_matchtrk_phi);
+  tree->SetBranchAddress("matchtrkExt_z0", &matchtrk_z0, &b_matchtrk_z0);
+  tree->SetBranchAddress("matchtrkExt_d0", &matchtrk_d0, &b_matchtrk_d0);
+  tree->SetBranchAddress("matchtrkExt_rinv", &matchtrk_rinv, &b_matchtrk_rinv);
+  tree->SetBranchAddress("matchtrkExt_chi2", &matchtrk_chi2, &b_matchtrk_chi2);
+  tree->SetBranchAddress("matchtrkExt_chi2rphi", &matchtrk_chi2rphi, &b_matchtrk_chi2rphi);
+  tree->SetBranchAddress("matchtrkExt_chi2rz", &matchtrk_chi2rz, &b_matchtrk_chi2rz);
+  tree->SetBranchAddress("matchtrkExt_bendchi2", &matchtrk_bendchi2, &b_matchtrk_bendchi2);
+  tree->SetBranchAddress("matchtrkExt_MVA", &matchtrk_MVA1, &b_matchtrk_MVA1);
+  tree->SetBranchAddress("matchtrkExt_MVA2", &matchtrk_MVA2, &b_matchtrk_MVA2);
+  tree->SetBranchAddress("matchtrkExt_nstub", &matchtrk_nstub, &b_matchtrk_nstub);
+  tree->SetBranchAddress("matchtrkExt_lhits", &matchtrk_lhits, &b_matchtrk_lhits);
+  tree->SetBranchAddress("matchtrkExt_dhits", &matchtrk_dhits, &b_matchtrk_dhits);
+  tree->SetBranchAddress("matchtrkExt_seed", &matchtrk_seed, &b_matchtrk_seed);
+  tree->SetBranchAddress("matchtrkExt_hitpattern", &matchtrk_hitpattern, &b_matchtrk_hitpattern);
+
+  //preselection cuts and plots definitions
+  // Cut assumptions: first cut is maxEta
+  std::vector<std::unique_ptr<Cut>> preselCuts;
+  std::unique_ptr<TypedCut<float>> cut0(new TypedCut<float>("maxEta","max #eta",&trk_eta,2.4,true));

```

Comment:
I know this is test code and isn't really intended for main CMSSW use, but `std::make_unique` is probably preferred.